pkg/segment/query/processor/inputlookupcommand_test.go

```diff
@@ -0,0 +1,306 @@
+	runningConfig.DataPath = "inputlookup-data/"
+	runningConfig.SSInstanceName = "test"
+	config.SetConfig(runningConfig)
+
+	return config.InitDerivedConfig("test")
+}
+
+func prepareData(data []string, fileName string) error {
+
+	err := os.MkdirAll(config.GetLookupPath(), os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	fd, err := os.Create(filepath.Join(config.GetLookupPath(), fileName))
+	if err != nil {
+		return err
+	}
+	defer fd.Close()
+
+	for _, d := range data {
+		_, err = fd.WriteString(d)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func Test_InputLookup(t *testing.T) {
+
+	err := initTestConfig(t)
+	assert.Nil(t, err)
+
+	data := []string{
+		"a,b,c\n",
+		"a1,b2,c3\n",
+		"a4,b5,c6\n",
+		"a7,b8,c9\n",
+	}
+
+	csvFile := "test.csv"
+
+	err = prepareData(data, csvFile)
+	assert.Nil(t, err)
+
+	dp := NewInputLookupDP(&structs.InputLookup{
+		IsFirstCommand: true,
+		Filename:       csvFile,
+		Start:          1,
+		Max:            3,
+	},
+	)
+
+	iqr, err := dp.Fetch()
+	assert.Nil(t, err)
+	assert.NotNil(t, iqr)
+	assert.Equal(t, 2, iqr.NumberOfRecords())
+
+	col_a, err := iqr.ReadColumn("a")
+	assert.Nil(t, err)
+	expected := []utils.CValueEnclosure{
+		{Dtype: utils.SS_DT_STRING, CVal: "a4"},
+		{Dtype: utils.SS_DT_STRING, CVal: "a7"},
+	}
+	assert.Equal(t, expected, col_a)
+
+	col_b, err := iqr.ReadColumn("b")
+	assert.Nil(t, err)
+	expected = []utils.CValueEnclosure{
+		{Dtype: utils.SS_DT_STRING, CVal: "b5"},
+		{Dtype: utils.SS_DT_STRING, CVal: "b8"},
+	}
+	assert.Equal(t, expected, col_b)
+
+	col_c, err := iqr.ReadColumn("c")
+	assert.Nil(t, err)
+	expected = []utils.CValueEnclosure{
+		{Dtype: utils.SS_DT_STRING, CVal: "c6"},
+		{Dtype: utils.SS_DT_STRING, CVal: "c9"},
+	}
+	assert.Equal(t, expected, col_c)
+
+	iqr, err = dp.Fetch()
+	assert.Equal(t, io.EOF, err)
+	assert.Nil(t, iqr)
+
+	os.RemoveAll(config.GetDataPath())
+}
+
+func getExpectedColumnValues(val string, start int, count int) []utils.CValueEnclosure {
+	var expected []utils.CValueEnclosure
+	for i := 0; i < count; i++ {
+		colValue := fmt.Sprintf("%v%v", val, start+i)
+		expected = append(expected, utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: colValue})
+	}
+	return expected
+}
+
+func validate_MultiFetch(t *testing.T, dp *DataProcessor, expectedCols map[string]struct{}, start []int, numRecords []int, numFetches int) {
+	fetch := 0
+	for {
+		iqr, err := dp.Fetch()
+		if fetch == numFetches {
+			assert.Equal(t, io.EOF, err)
+			break
+		} else {
+			assert.Nil(t, err)
+		}
+		assert.NotNil(t, iqr)
+		assert.Equal(t, numRecords[fetch], iqr.NumberOfRecords())
+		actualCols, err := iqr.GetColumns()
+		assert.Nil(t, err)
+		assert.Equal(t, expectedCols, actualCols)
+
+		// Column values from 2 to 101 would be read
+		for col := range expectedCols {
+			colValues, err := iqr.ReadColumn(col)
+			assert.Nil(t, err)
+			assert.Equal(t, getExpectedColumnValues(col, start[fetch], numRecords[fetch]), colValues)
+		}
```

Comment:
I'm confused. Where are 2 and 101 coming from?

Reply:
forgot to remove the comment, addressed in the latest commit.